common/cmake/devices.cmake

```diff
@@ -1118,15 +1118,15 @@ function(ADD_FPGA_TARGET)
 
   set(YOSYS_IO_DEPS "")
 
-  if(NOT ${ADD_FPGA_TARGET_INPUT_IO_FILE} STREQUAL "")
+  if(NOT ${ADD_FPGA_TARGET_INPUT_IO_FILE} STREQUAL "" OR NOT ${ADD_FPGA_TARGET_INPUT_XDC_FILE} STREQUAL "")
     get_target_property_required(PINMAP_FILE ${BOARD} PINMAP)
-
-    get_file_location(INPUT_IO_FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
     get_file_location(PINMAP ${PINMAP_FILE})
-
-    append_file_dependency(YOSYS_IO_DEPS ${ADD_FPGA_TARGET_INPUT_IO_FILE})
     append_file_dependency(YOSYS_IO_DEPS ${PINMAP_FILE})
+  endif()
 
+  if(NOT ${ADD_FPGA_TARGET_INPUT_IO_FILE} STREQUAL "")
+    get_file_location(INPUT_IO_FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
+    append_file_dependency(YOSYS_IO_DEPS ${ADD_FPGA_TARGET_INPUT_IO_FILE})
```

Comment:
You need to call `append_file_dependency(YOSYS_IO_DEPS ${ADD_FPGA_TARGET_INPUT_XDC_FILE})` if it is specified

Reply:
`INPUT_XDC_FILE` has been already added to the dependencies list of the command that has `YOSYS_IO_DEPS`, but I guess it will be cleaner with `append_file_dependency`.
I still need to work on the PCF backward compatibility in the ioplace python script so currently CI will fail.

Reply:
So append_file_dependency does more than just add the file location, it adds the file target as well.  Both are required for proper dependency management.  One without the other does not work as expected if the input file is sourced from a different working directory and is a generated file.